### PR TITLE
Cities dropdown

### LINKED ISSUE #454: Countries should appear as dropdowns in volunteer search and sign up forms
Right now the users have to manually type the country in the fields. It'd be great if users can select country through a drop down list. In that case, even validation won't be required. Also, this is implemented already in systers/portal which vms is going to be a part of (correct me if i'm wrong :smile:)

more of an enhancement, IMO.

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
### Description
Django cities light package implemented thus countries, cities and states appearing as drop down.

Fixes #454 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

How Has This Been Tested?
image
image
image

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 @nida @kamsuri The previous PR accidentally got reset, and I am not able to retrieve it back, so Please review this one

File: vms/event/views.py [hunk 1]
```diff
@@ -48,17 +49,43 @@ class EventCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin,
...
+                'country_list': country_list,
             })
         else:
-            form.save()
+            event = form.save(commit=False)
+            try:
+                country_id = self.request.POST.get('country')
+                country = Country.objects.get(pk=country_id)
+                event.country = country
+            except:
```
Comment (Monal5031) [inline vms/event/views.py]: bare except should never be used. Use at least `Exception`

File: vms/event/views.py [hunk 2]
```diff
@@ -48,17 +49,43 @@ class EventCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin,
...
+                country_id =None
```
Comment (Monal5031) [inline vms/event/views.py]: Space around binary operator.

File: vms/event/views.py [hunk 3]
```diff
@@ -48,17 +49,43 @@ class EventCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin,
...
+            try:
+                state_id = self.request.POST.get('state')
+                state = Region.objects.get(pk=state_id)
+                event.state = state
+            except:
```
Comment (Monal5031) [inline vms/event/views.py]: Same here.

File: vms/event/views.py [hunk 4]
```diff
@@ -48,17 +49,43 @@ class EventCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin,
...
+                state_id = None
+            try:
+                city_id = self.request.POST.get('city')
+                city = City.objects.get(pk=city_id)
+                event.city = city
+            except:
```
Comment (Monal5031) [inline vms/event/views.py]: Same here.

File: vms/event/views.py [hunk 5]
```diff
@@ -48,17 +49,43 @@ class EventCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin,
...
+                state_id = None
```
Comment (Monal5031) [inline vms/event/views.py]: `city_id` :stuck_out_tongue:

File: vms/event/views.py [hunk 6]
```diff
@@ -120,6 +147,15 @@ def get_context_data(self, **kwargs):
...
                                                   'end_date').distinct()
+        context['country_list'] = Country.objects.all()
+        event_id = self.kwargs['event_id']
+        event =  get_event_by_id(event_id)
+        context['country'] = event.country
+        context['country_id'] = event.country.pk
+        context['state'] = event.state
+        context['state_id'] = event.state.pk
+        context['city'] = event.city
+        context['city_id'] = event.city.pk
```
Comment (Monal5031) [inline vms/event/views.py]: I don't know if in future we implement the reuse of id will affect this or not. can you please check it once.

File: vms/registration/views.py [hunk 7]
```diff
@@ -26,6 +26,8 @@
 from registration.tokens import account_activation_token
 from volunteer.forms import VolunteerForm
 from volunteer.validation import validate_file
+from volunteer.models import Volunteer
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/registration/views.py]: Codacy's warning.

File: vms/registration/views.py [hunk 8]
```diff
@@ -57,10 +60,13 @@ def get(self, request):
                 'phone_error': self.phone_error,
                 'match_error': self.match_error,
                 'organization_list': self.organization_list,
+                'country_list': self.country_list,
             })
 
     def post(self, request):
         organization_list = get_organizations_ordered_by_name()
+        country_list = Country.objects.all()
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/registration/views.py]: codacy points this as unused.

File: vms/registration/views.py [hunk 9]
```diff
@@ -82,10 +88,26 @@ def post(self, request):
                                 'match_error': self.match_error,
                                 'organization_list': self.organization_list,
                             })
-                    ad_country = request.POST.get('admin-country')
-                    ad_phone = request.POST.get('admin-phone_number')
+                    try:
+                        ad_country_id = request.POST.get('country')
+                        ad_country = Country.objects.get(pk=ad_country_id)
+                    except:
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/registration/views.py]: bare except here too.

File: vms/registration/views.py [hunk 10]
```diff
@@ -82,10 +88,26 @@ def post(self, request):
...
+                        ad_country = None
+
+                    try:
+                        ad_state_id = request.POST.get('state')
+                        ad_state = Region.objects.get(pk=ad_state_id)
+                    except:
```
Comment (Monal5031) [inline vms/registration/views.py]: Bare except.

File: vms/registration/views.py [hunk 11]
```diff
@@ -82,10 +88,26 @@ def post(self, request):
...
+                        ad_state = None
+
+                    try:
+                        ad_city_id = request.POST.get('city')
+                        ad_city = City.objects.get(pk=ad_city_id)
+                    except:
```
Comment (Monal5031) [inline vms/registration/views.py]: bare except.

File: vms/registration/views.py [hunk 12]
```diff
@@ -82,10 +88,26 @@ def post(self, request):
                                 'match_error': self.match_error,
                                 'organization_list': self.organization_list,
                             })
-                    ad_country = request.POST.get('admin-country')
-                    ad_phone = request.POST.get('admin-phone_number')
+                    try:
+                        ad_country_id = request.POST.get('country')
```
Comment (Monal5031) [inline vms/registration/views.py]: and make all these `admin` instead of `ad` please.

File: vms/registration/views.py [hunk 13]
```diff
@@ -175,8 +210,24 @@ def post(self, request):
                                 'match_error': self.match_error,
                                 'organization_list': self.organization_list,
                             })
+                    try:
+                        vol_country_id = request.POST.get('country')
+                        vol_country = Country.objects.get(pk=vol_country_id)
+                    except:
```
Comment (Monal5031) [inline vms/registration/views.py]: bare except.

File: vms/registration/views.py [hunk 14]
```diff
@@ -175,8 +210,24 @@ def post(self, request):
...
+                    try:
+                        vol_state_id = request.POST.get('state')
+                        vol_state = Region.objects.get(pk=vol_state_id)
+                    except:
+                        vol_state = None
+
+                    try:
+                        vol_city_id = request.POST.get('city')
+                        vol_city = City.objects.get(pk=vol_city_id)
+                    except:
```
Comment (Monal5031) [inline vms/registration/views.py]: bare except.

File: vms/registration/views.py [hunk 15]
```diff
@@ -175,8 +210,24 @@ def post(self, request):
...
+                    try:
+                        vol_country_id = request.POST.get('country')
+                        vol_country = Country.objects.get(pk=vol_country_id)
+                    except:
+                        vol_country = None
+
+                    try:
+                        vol_state_id = request.POST.get('state')
+                        vol_state = Region.objects.get(pk=vol_state_id)
+                    except:
```
Comment (Monal5031) [inline vms/registration/views.py]: bare except.

File: vms/registration/views.py [hunk 16]
```diff
@@ -199,11 +251,12 @@ def post(self, request):
                                 request, 'registration/signup_volunteer.html',
                                 {
                                     'user_form': user_form,
-                                    'volunteer_form': volunteer_form,
+                                    'volunteer_fo-rm': volunteer_form,
```
Comment (Monal5031) [inline vms/registration/views.py]: 'volunteer_fo-rm' :thinking:

File: vms/registration/views.py [hunk 17]
```diff
@@ -271,3 +332,40 @@ def activate(request, uidb64, token):
     else:
         return HttpResponseBadRequest('Activation link is invalid!')
 
+def check_states(request):
+    """
+    check if states exist in a country
```
Comment (Monal5031) [inline vms/registration/views.py]: 1st letter caps

File: vms/registration/views.py [hunk 18]
```diff
@@ -271,3 +332,40 @@ def activate(request, uidb64, token):
...
+    country_id = request.GET.get('country')
+    if Region.objects.filter(country_id=country_id).exists():
+       statecheck = 1
+    else:
+       statecheck = 0
+    return JsonResponse(statecheck, safe=False)
+
+def load_states(request):
+    """
+    renders the options of states dropdown list
```
Comment (Monal5031) [inline vms/registration/views.py]: 1st letter caps.

File: vms/registration/views.py [hunk 19]
```diff
@@ -271,3 +332,40 @@ def activate(request, uidb64, token):
...
+
+    :return: states belonging to the selected country
+    """
+    country_id = request.GET.get('country')
+    states = Region.objects.filter(country_id=country_id).order_by('name')
+    return render(request, 'registration/state_dropdown_list_options.html',{'states':states})
+
+def load_cities(request):
+    """
+    renders the options of cities dropdown
```
Comment (Monal5031) [inline vms/registration/views.py]: 1st letter caps.

File: vms/shift/utils.py [hunk 20]
```diff
@@ -190,6 +190,32 @@ def create_country():
...
+        name_ascii = "Uttarakhand",
+        slug='uttarakhand',
+        geoname_id='1444366',
+        alternate_names='',
+        name='Uttarakhand',
+        geoname_code='39',
+        country=country)
+
+
+def create_city():
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/shift/utils.py]: codacy's error for whitespace at end.

File: vms/shift/utils.py [hunk 21]
```diff
@@ -190,6 +190,32 @@ def create_country():
...
+def create_state():
+    country = Country.objects.get(name='India')
+    Region.objects.create(
+        name_ascii = "Uttarakhand",
+        slug='uttarakhand',
+        geoname_id='1444366',
+        alternate_names='',
+        name='Uttarakhand',
+        geoname_code='39',
+        country=country)
```
Comment (Monal5031) [inline vms/shift/utils.py]: end bracket on next line

File: vms/volunteer/models.py [hunk 22]
```diff
@@ -96,6 +82,5 @@ class Volunteer(models.Model):
         blank=True)
 
     user = models.OneToOneField(User)
-
```
Comment (Monal5031) [inline vms/volunteer/models.py]: we need this blank line

File: vms/volunteer/services.py [hunk 23]
```diff
@@ -7,7 +7,6 @@
                                    get_organizations_ordered_by_name)
 from volunteer.models import Volunteer
 
-
```
Comment (Monal5031) [inline vms/volunteer/services.py]: This one too.

File: vms/event/views.py (hunk 6 above)
Comment (anjali-dhanuka) [reply vms/event/views.py]: These ids have nothing to do with VMS , they are of cities light. I don't think that it wil affect

**Comment by kamsuri** [comment]
@anjali-dhanuka why are all the tests failing?

File: vms/event/tests/test_searchEvent.py [hunk 24]
```diff
@@ -439,14 +457,14 @@ def test_intersection_of_all_fields(self):
...
-        search_page.search_country_field('event')
+        search_page.search_state_field('Uttarakhand')
+        search_page.search_country_field('India')
         search_page.search_job_field('job')
         search_page.submit_form()
         search_results = search_page.get_search_results()
         result = search_page.get_results_list(search_results)
         self.assertTrue(expected_result_one in result)
-        self.assertTrue(expected_result_two in result)
+
```
Comment (Monal5031) [inline vms/event/tests/test_searchEvent.py]: extra line

File: vms/pom/pages/adminRegistrationPage.py [hunk 25]
```diff
@@ -19,6 +22,19 @@ def __init__(self, driver):
...
 
+    def ajax_complete(driver):
+        try:
+            return 0 == driver.execute_script("return jQuery.active")
+        except WebDriverException:
+            pass
+
+    def _wait_ajax_complete(self):
+        """ Wait until ajax request is completed """
+        driver = webdriver.Firefox()
```
Comment (Monal5031) [inline vms/pom/pages/adminRegistrationPage.py]: btw this is the reason for multiple firefox opening up :stuck_out_tongue:

**Comment by anjali-dhanuka** [comment]
Hi @abhishekarya286 @nida @kamsuri Please review

**Comment by anjali-dhanuka** [comment]
All the tests are passing @kamsuri

**Comment by abhishekarya286** [comment]
Hi @anjali-dhanuka. Please resolve conflicts.

File: vms/administrator/templates/administrator/edit.html [hunk 26]
```diff
@@ -113,84 +113,72 @@
...
+                    <option value="{{ country.name }}">{{ country.name }}</option>
+                {% endif %}
+                {% else %}
+                <option value="{{ country.name }}">{{ country.name }}</option>
+                {% endif %}
+                {% endfor %}
+                </select>
+                </div>
+                </div>
+
```
Comment (abhishekarya286) [inline vms/administrator/templates/administrator/edit.html]: Extra line.

File: vms/administrator/templates/administrator/edit.html [hunk 27]
```diff
@@ -113,84 +113,72 @@
...
+                <option value="0">-- {% trans "Select state" %} --</option>
+                {% for state in state_list %}
+                 {% if administrator.state_id %}
+                     {% if state.name == administrator.state.name %}
+                         <option selected value="{{state.id}}">{{state.name}}</option>
+                     {% else %}
+                    <option value="{{ state.name }}">{{ state.name }}</option>
+                {% endif %}
+                {% else %}
+                <option value="{{ state.name }}">{{ state.name }}</option>
```
Comment (abhishekarya286) [inline vms/administrator/templates/administrator/edit.html]: indent it.

File: vms/event/templates/event/create.html [hunk 28]
```diff
@@ -89,76 +89,41 @@
...
+        <div class="form-group">
+        <label class="col-md-2 control-label">
+        {% trans "City" %}
+        </label>
+        <div class="col-md-8">
+        <select id="select_city" class="form-control" name="city">
+            </select>
+            </div>
+            </div>
+
```
Comment (abhishekarya286) [inline vms/event/templates/event/create.html]: Extra line.

File: vms/event/templates/event/create.html [hunk 29]
```diff
@@ -181,7 +146,7 @@
                         </div>
                     </div>
                 {% endif %}
-				
+
 				{% if form.venue.errors %}
```
Comment (abhishekarya286) [inline vms/event/templates/event/create.html]: Correct indentation here.

File: vms/event/templates/event/create.html [hunk 30]
```diff
@@ -204,8 +169,8 @@
                         </div>
                     </div>
                 {% endif %}
-				
-				
+
```
Comment (abhishekarya286) [inline vms/event/templates/event/create.html]: extra line

File: vms/event/templates/event/edit.html [hunk 31]
```diff
@@ -100,76 +100,74 @@
...
+            <div id="div_id_state" class="form-group">
+                <label class="col-md-2 control-label">
+                    {% trans "State/Province" %}
+                </label>
+                <div class="col-md-8">
+                <select id="select_state" class="form-control" name="state">
+
+                <option value="0">-- {% trans "Select state" %} --</option>
+                {% for state in state_list %}
+                 {% if event.state_id %}
```
Comment (abhishekarya286) [inline vms/event/templates/event/edit.html]: make indentation uniform here. Somewhere it is too much and somewhere it's two less.

**Comment by anjali-dhanuka** [comment]
@kamsuri @abhishekarya286 @nida @Monal5031 It is ready for review, the failing test will be fixed as soon as @Monal5031 is done with the dependency issue

File: vms/administrator/tests/test_settings.py [hunk 32]
```diff
@@ -10,7 +10,7 @@
 from pom.pages.authenticationPage import AuthenticationPage
 from pom.pages.jobDetailsPage import JobDetailsPage
 from pom.locators.eventsPageLocators import EventsPageLocators
-from shift.utils import ( create_admin_with_unlisted_org, 
+from shift.utils import ( create_admin_with_unlisted_org,
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/administrator/tests/test_settings.py]: unused import `create_admin`.

File: vms/event/views.py [hunk 33]
```diff
@@ -234,9 +291,18 @@ def return_event_data(self, events):
             event_data['end_date'] = event.end_date
             event_data['description'] = event.description
             event_data['address'] = event.address
-            event_data['city'] = event.city
-            event_data['state'] = event.state
-            event_data['country'] = event.country
+            if event.city:
```
Comment (Monal5031) [inline vms/event/views.py]: This can be one liner

File: vms/event/views.py [hunk 34]
```diff
@@ -234,9 +291,18 @@ def return_event_data(self, events):
...
+                event_data['city'] = event.city.name
+            else:
+                event_data['city'] = None
+            if event.state:
```
Comment (Monal5031) [inline vms/event/views.py]: This can be one liner

File: vms/event/views.py [hunk 35]
```diff
@@ -234,9 +291,18 @@ def return_event_data(self, events):
...
+                event_data['state'] = event.state.name
+            else:
+                event_data['state'] = None
+            if event.country:
```
Comment (Monal5031) [inline vms/event/views.py]: This can be one liner

File: vms/pom/pages/adminRegistrationPage.py [hunk 36]
```diff
@@ -28,9 +28,9 @@ def fill_registration_form(self, info):
         self.send_value_to_element_id(elements.LAST_NAME, info[4])
         self.send_value_to_element_id(elements.EMAIL, info[5])
         self.send_value_to_element_id(elements.ADDRESS, info[6])
-        self.send_value_to_element_id(elements.CITY, info[7])
-        self.send_value_to_element_id(elements.STATE, info[8])
         self.send_value_to_element_id(elements.COUNTRY, info[9])
+        self.send_value_to_element_id(elements.STATE, info[8])
```
Comment (Monal5031) [inline vms/pom/pages/adminRegistrationPage.py]: Why this change?

File: vms/pom/pages/volunteerProfilePage.py [hunk 37]
```diff
@@ -23,9 +23,9 @@ def fill_values(self, new_details):
         self.fill_field(elements.PROFILE_LAST_NAME, new_details[1])
         self.fill_field(elements.PROFILE_EMAIL, new_details[2])
         self.fill_field(elements.PROFILE_ADDRESS, new_details[3])
-        self.fill_field(elements.PROFILE_CITY, new_details[4])
-        self.fill_field(elements.PROFILE_STATE, new_details[5])
-        self.fill_field(elements.PROFILE_COUNTRY, new_details[6])
+        self.send_value_to_xpath(elements.PROFILE_COUNTRY, new_details[6])
```
Comment (Monal5031) [inline vms/pom/pages/volunteerProfilePage.py]: Why change the order? Its anyhow going to be converted to dictionaries in my next week.

File: vms/registration/views.py [hunk 38]
```diff
@@ -27,6 +28,7 @@
 from registration.tokens import account_activation_token
 from volunteer.forms import VolunteerForm
 from volunteer.validation import validate_file
+from cities_light.models import City, Region, Country
```
Comment (Monal5031) [inline vms/registration/views.py]: sorted import.

File: vms/registration/views.py [hunk 39]
```diff
@@ -156,10 +188,13 @@ def get(self, request):
                 'phone_error': self.phone_error,
                 'match_error': self.match_error,
                 'organization_list': self.organization_list,
+                'country_list': self.country_list,
             })
 
-    def post(self, request):
+    def post(self,request):
```
Comment (Monal5031) [inline vms/registration/views.py]: space after comma

File: vms/registration/views.py [hunk 40]
```diff
@@ -231,8 +284,12 @@ def post(self, request):
                         org = Organization.objects.create(name=unlisted_org, approved_status=False)
                         org.save()
                         volunteer.organization = org
-
-                    volunteer.reminder_days = 1
+                    if vol_country:
+                        volunteer.country = vol_country
```
Comment (Monal5031) [inline vms/registration/views.py]: shouln't there be else where we assign none.

File: vms/registration/views.py [hunk 41]
```diff
@@ -281,3 +339,40 @@ def activate(request, uidb64, token):
...
         return HttpResponseBadRequest('Activation link is invalid!')
 
+def check_states(request):
+    """
+    Check if states exist in a country
+
+    :return: 1 if states exist, otherwise 0
+    """
+    country_name = request.GET.get('country')
+    if Region.objects.filter(country__name=country_name).exists():
```
Comment (Monal5031) [inline vms/registration/views.py]: This can be one liner

File: vms/registration/views.py [hunk 42]
```diff
@@ -281,3 +339,40 @@ def activate(request, uidb64, token):
...
+    Renders the options of cities dropdown
+
+    :return: cities belonging to the selected country and state
+    """
+    country_name = request.GET.get('country')
+    state_name = request.GET.get('state')
+    if state_name is '0':
+        cities = City.objects.filter(country__name=country_name).order_by('name')
+    else:
+        cities = City.objects.filter(country__name=country_name,region__name=state_name).order_by('name')
```
Comment (Monal5031) [inline vms/registration/views.py]: space after comma

File: vms/shift/tests/test_services.py [hunk 43]
```diff
@@ -18,7 +18,11 @@
     get_volunteer_shifts, get_volunteer_shifts_with_hours, get_volunteers_by_shift_id,
     get_logged_volunteers_by_shift_id, is_signed_up, register, send_reminder,
     get_shifts_with_open_slots_for_volunteer, get_report_by_id)
-from shift.utils import create_event_with_details, create_job_with_details, create_organization_with_details, create_shift_with_details, create_report_with_details, clear_objects, get_report_list, create_volunteer_with_details, register_event_utility, register_job_utility, register_shift_utility,  set_shift_location, log_hours_with_details
+from shift.utils import (create_report_with_details, create_organization_with_details,
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/shift/tests/test_services.py]: unused `create_second_country`

File: vms/shift/tests/test_shiftDetails.py [hunk 44]
```diff
@@ -10,10 +10,8 @@
 # local Django
 from pom.pages.authenticationPage import AuthenticationPage
 from pom.pages.shiftDetailsPage import ShiftDetailsPage
-from shift.utils import (create_volunteer_with_details, create_admin,
-                         create_event_with_details, create_job_with_details,
-                         create_organization_with_details,
-                         create_shift_with_details, log_hours_with_details,
+from shift.utils import (create_volunteer, create_admin, create_event_with_details, create_job_with_details,
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/shift/tests/test_shiftDetails.py]: unused `create_organization_with_details`

File: vms/volunteer/tests/test_searchVolunteer.py [hunk 45]
```diff
@@ -270,56 +279,64 @@ def test_volunteer_city_field(self):
...
         search_page.submit_form()
         search_results = search_page.get_search_results()
         result = search_page.get_results_list(search_results)
-        self.assertEqual(len(result), 2)
-
-        self.assertTrue(expected_result_one in result)
+        self.assertEqual(len(result), 1)
         self.assertTrue(expected_result_two in result)
-
+
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/volunteer/tests/test_searchVolunteer.py]: trailing whitespace.

File: vms/volunteer/tests/test_services.py [hunk 46]
```diff
@@ -3,8 +3,8 @@
 
 # local Django
 from organization.models import Organization
-from shift.utils import (create_volunteer_with_details, create_organization_with_details, clear_objects,
-    register_event_utility, register_job_utility, register_shift_utility,
+from shift.utils import (create_second_state, create_second_country, create_second_city, create_volunteer_with_details, create_organization_with_details, clear_objects,
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/volunteer/tests/test_services.py]: unused `create_second_state`.

File: vms/volunteer/tests/test_unit.py [hunk 47]
```diff
@@ -133,7 +140,7 @@ def test_model_edit_with_invalid_values(self):
         self.assertEqual(valid_volunteer.phone_number, volunteer_in_db.phone_number)
 
         valid_volunteer.first_name = 'Prince'
-        valid_volunteer.last_name = 'Vegeta'
+        valid_volunteer.last_name = 'Vegeta'
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/volunteer/tests/test_unit.py]: trailing whitespace.

File: vms/volunteer/templates/volunteer/edit.html [hunk 48]
```diff
@@ -121,72 +121,69 @@
...
-                        <div class="col-md-8">
-                            <input class="form-control" type="text" placeholder="{% blocktrans %}State/Province{% endblocktrans %}" name="state" value="{% if form.state.value %}{{ form.state.value }}{% endif %}">
-                        </div>
+                <div id="div_id_country" class="form-group">
+                    <label class="col-md-2 control-label">
+                    {% trans "Country" %}
+                    </label>
+                    <div class="col-md-8">
+                        <select id="select_country" class="form-control" name="country">
+                            <option value="0">-- {% trans "Select country" %} --</option>
```
Comment (abhishekarya286) [inline vms/volunteer/templates/volunteer/edit.html]: indentation??

File: vms/volunteer/templates/volunteer/edit.html [hunk 49]
```diff
@@ -121,72 +121,69 @@
...
-                            </p>
-                        </div>
+                </div>
+                <div id="div_id_state" class="form-group">
+                    <label class="col-md-2 control-label">
+                    {% trans "State/Province" %}
+                    </label>
+                    <div class="col-md-8">
+                        <select id="select_state" class="form-control" name="state">
+                            <option value="0">-- {% trans "Select state" %} --</option>
```
Comment (abhishekarya286) [inline vms/volunteer/templates/volunteer/edit.html]: same here

File: vms/volunteer/templates/volunteer/edit.html [hunk 50]
```diff
@@ -121,72 +121,69 @@
...
-                            <input class="form-control" type="text" placeholder="{% blocktrans %}Country{% endblocktrans %}" name="country" value="{% if form.country.value %}{{ form.country.value }}{% endif %}">
-                        </div>
+                </div>
+                <div id="div_id_city" class="form-group">
+                    <label class="col-md-2 control-label">
+                    {% trans "City" %}
+                    </label>
+                    <div class="col-md-8">
+                        <select id="select_city" class="form-control" name="city">
+                            <option value="0">-- {% trans "Select city" %} --</option>
```
Comment (abhishekarya286) [inline vms/volunteer/templates/volunteer/edit.html]: same here

File: vms/event/tests/test_shiftSignUp.py [hunk 51]
```diff
@@ -172,10 +173,14 @@ def test_shift_sign_up_with_no_slots(self):
         s2 = create_shift_with_details(shift_2)
 
         # Create another volunteer
-        volunteer_2 = ['volunteer-2', "Sam", "Turtle", "Mario Land", "Nintendo Land", "Nintendo State",
-                       "Nintendo Nation", "2374983247", "volunteer2@volunteer.com"]
         org_name = 'volunteer-organization'
         org_obj = create_organization_with_details(org_name)
+
+        second_country = create_second_country()
```
[the remark this answers is not in the document]
Comment (anjali-dhanuka) [reply vms/event/tests/test_shiftSignUp.py]: There are other arguments also which are required for creating a country, namely geoname id,  phone, continent, asci name and many more, wihich can't be passed every time.

File: vms/pom/pages/adminRegistrationPage.py (hunk 36 above)
Comment (anjali-dhanuka) [reply vms/pom/pages/adminRegistrationPage.py]: To resemble that the order is chnaged in which we fill the form.

File: vms/pom/pages/volunteerProfilePage.py (hunk 37 above)
Comment (anjali-dhanuka) [reply vms/pom/pages/volunteerProfilePage.py]: Answered above

File: vms/registration/views.py (hunk 40 above)
Comment (anjali-dhanuka) [reply vms/registration/views.py]: I think even this if is not reqired, I have already used if else above.

**Comment by abhishekarya286** [comment]
Please resolve the conflicts.

File: vms/event/tests/test_services.py [hunk 52]
```diff
@@ -23,7 +22,11 @@ def setUpModule():
     global e1, e2, e3, e4, e5
```
[the remark this answers is not in the document]
Comment (abhishekarya286) [reply vms/event/tests/test_services.py]: ??

File: vms/event/tests/test_shiftSignUp.py (hunk 51 above)
[the remark this answers is not in the document]
Comment (abhishekarya286) [reply vms/event/tests/test_shiftSignUp.py]: okay. No problem then.

File: vms/registration/templates/registration/signup_administrator.html [hunk 53]
```diff
@@ -206,84 +206,70 @@ <h2>{% trans "Sign Up" %}</h2>
...
-                    <input id="id_country" class="textinput textInput form-control" type="text" placeholder="{% blocktrans %}Country{% endblocktrans %}" name="admin-country" value="{% if administrator_form.country.value %}{{ administrator_form.country.value }}{% endif %}">
-                </div>
+        </div>
+        <div id="div_id_city" class="form-group">
+            <label class="col-md-2 control-label">
+            {% trans "City" %}
+            </label>
+            <div class="col-md-8">
+                <select id="select_city" class="form-control" name="city">
+                    <option value="0">-- {% trans "Select city" %} --</option>
```
[the remark this answers is not in the document]
Comment (abhishekarya286) [reply vms/registration/templates/registration/signup_administrator.html]: please indent here.

File: vms/shift/utils.py [hunk 54]
```diff
@@ -200,34 +214,127 @@ def create_country():
         continent='AS',
         tld='in',
         phone='91')
+    return country
+
```
Comment (abhishekarya286) [inline vms/shift/utils.py]: Add another blank line

File: vms/shift/utils.py [hunk 55]
```diff
@@ -200,34 +214,127 @@ def create_country():
...
+        name_ascii = "Uttarakhand",
+        slug='uttarakhand',
+        geoname_id='1444366',
+        alternate_names='',
+        name='Uttarakhand',
+        geoname_code='39',
+        country=country
+        )
+    return state
+
```
Comment (abhishekarya286) [inline vms/shift/utils.py]: same here

**Comment by abhishekarya286** [comment]
Fix the conflicts first.
